aws-api/src/main/java/com/amplifyframework/api/aws/GsonGraphQLResponseFactory.java

```diff
@@ -124,7 +125,11 @@
                 jsonData.getAsJsonObject().has(ITEMS_KEY)
         ) {
             Iterable<T> data = parseDataAsList(jsonData.getAsJsonObject().get(ITEMS_KEY), classToCast);
-            return new GraphQLResponse<>(data, errors);
+            String nextToken = null;
+            if(jsonData.getAsJsonObject().has(NEXT_TOKEN_KEY) && jsonData.getAsJsonObject().get(NEXT_TOKEN_KEY).isJsonPrimitive()) {
+                nextToken = jsonData.getAsJsonObject().get(NEXT_TOKEN_KEY).getAsString();
+            }
+            return new GraphQLResponse<>(data, errors, nextToken);
```

Comment:
Maybe move this up inside `if`, after the `nextToken` assignment`, so that you know the `nextToken` is not `null`.

Then, as a default, you could just
```
return new GraphQLResponse<>(data, errors);
```
(with no `null`-valued `nextToken` specified.)